src/Framework/Logging/AnsiCodes.cs

```diff
@@ -122,6 +124,11 @@ internal static class AnsiCodes
     /// </remarks>
     public const string RemoveProgress = "\x1b]9;4;0;\x1b\\";
 
+    /// <summary>
+    /// Remove all ANSI escape codes from a string.
+    /// </summary>
+    private static readonly Regex RemoveAnsiRegex = new Regex(@"\x1B\[[^@-~]*[@-~]", RegexOptions.Compiled);
```

Comment:
This will inspect every target on update, maybe this is too heavy and we should pass some structure that describes the output and colorization rather than reparsing ANSI out of string?

Reply:
solved